### ISSUE TITLE
Design left-handed mode

### ISSUE THREAD

**Issue by severinrudie** [mozilla-mobile/fenix#6365]
### Why/User Benefit/User Problem
The current design places the most important actions near the bottom right corner of the screen.  This is very convenient for right-handed users, but not for left-handed ones.

### Engineering note
Once we support RTL layout, providing an option to toggle it regardless of locale would be a relatively small engineering story.  This is one potential solution, but there may be better ones.

### Acceptance Criteria (how do I know when I’m done?)
- A design has been created that accommodates left-handed users

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FNXV2-14199)

**Comment by brampitoyo** [comment]
@Baron-Severin 

**TL;DR** This isn’t an issue, because people are adaptable.

Actually, as a left-hander, I really feel the pain – have been for most of my life.

Unfortunately the RTL layout turns things too drastically the other way round:

1. Good: most actions will be positioned near the bottom left hand corner of the screen
2. Bad: my language is LTR
    * My Back direction is left and my Forward direction is right; RTL layout turns it around
    * I read from left to right; RTL layout makes all text right-aligned

So, I want some things positioned LTR and other things RTL. Unlike an RTL language where everything is right-to-left.

Additionally: most RTL users are right-handed, but when they use the RTL layout, won’t the main menu be flipped and located on the left hand side of the screen, away from their thumbs?

So, how is handedness different from language orientation, and how are they the same? And should we give choices for both?

I’m proposing that perhaps we should pick only one standard to adhere to, and that standard should be **language orientation**.

Why not adhere to handedness? Read https://ux.stackexchange.com/questions/346/left-handed-persons-and-usability


> …a touchscreen device - especially a mobile, one-hand operated touchscreen device - is not always operated by means of the user's dominant hand…

Left-handed people sometimes hold their phones with their right hand – vice versa. This makes a right-handed layout useable.

> We don't make left handed door knobs, or left handed stick shifts.

This argument is a little shaky. In the physical world, the status quo may not be the way we like it. But onscreen, why can’t we change things?

This is why.

> …many left handed people work many things right handed way. Surely, that develops a mental model […] user get used to of the layout […] their mental models are set to expect things like a right handed person would expect.

Left-handed people (like me) live in a world full of, and optimised for, right-handedness. Some fight and use left-handed devices, but many adapted by simply learning how operate like right-handed people.

For better or worse, then, left-handed people come to expect things to operate in a right-handed way, _even though it conflicts with their natural inclination_. We survived, and for some of us, it didn’t end up becoming a big deal.

**Comment by saschanaz** [comment]
>TL;DR This isn’t an issue, because people are adaptable.

This is not a best reasoning. It should really be some kind of "too hard to design/implement" instead, because adaptable does not mean it's already ideal. Adapting is just the only choice when there is no alternative, and it's not a happy experience.

>I’m proposing that perhaps we should pick only one standard to adhere to, and that standard should be language orientation.

If Fenix can change the layout based on language orientation, why not allowing users to force the change? The layout might not be ideally pretty, but IMO usability should be preferred.

>Left-handed people (like me) live in a world full of, and optimised for, right-handedness. 

200% true. And please don't strengthen it.

What I want here is allowing to move the tab button and three-dot button to the left. IMO it should be sufficient for left handed users and shouldn't require too many sophisticated rationales.

**Comment by TheDecryptor** [comment]
I'd love a left handed mode, even though I'm right handed. I usually hold my phone in my left hand and type with my right (Either on the phone itself, or on a computer while I'm reading from my phone), and it can be rather awkward to reach to the far corners of the screen.

**Comment by mdesson** [comment]
To add to this, as a person with smaller hands in a world with increasingly larger phones, reaching the lower-right corner is an impossibility for me one-handed. The QoL change from this would be immense for me as a user. I can't really adapt to it without wrist strain (something I'm trying to reduce as a developer) or using my other hand (which defeats the convenience of putting the toolbar at the bottom in the first place since I'd be using it two-handed).

**Comment by nixin72** [comment]
As a left-handed person, having a left mode would be incredibly helpful. I find it very awkward and uncomfortable using my right hand with my phone since I just don't have good coordination. Telling left-handed people to "just adapt" is a little unfair in my opinion. 

It's also a matter of accessibility for people missing their right arm or have limited movement or something in their right arm. A left handed mode should definitely be added to accommodate lots of different people. Even if simply unfair to tell lefties to just adapt, people that fall into this category outright can't.

**Comment by AmyYLee** [comment]
@brampitoyo for follow-up

**Comment by LittleAngelwings** [comment]
Now for the new tab tray, it will be necessary. But also with the old tab screen. It's hard to reach the right button with the thumb for me as a left handed person with small hands. Please implement an option for left handed as an a11y feature.

**Comment by brampitoyo** [comment]
@Baron-Severin Now that support for RTL layout has landed, do you think that we can easily enable left-handed mode?

Obviously, we can’t simply turn RTL on and keep the language LTR. There are quite a lot of behaviour fixes we’ll need to make.

Left-handed mode takes a mixture of RTL and LTR layout:

|  | RTL layout  | LTR layout | Left-handed mode |
| --- | --- | --- | --- |
| Icon and text alignment in all screens, menus, dialogs, etc. | Right-aligned, text goes to the left of icons | Left-aligned, text goes to the right of icons | Follow LTR |
| Ordering of toolbar items (left-to-right) | Main menu, Tab view, Address bar | Address bar, Tab view, Main menu | Follow RTL |
| Main menu icon alignment | Left corner | Right corner | Follow RTL |
| Ordering of top app bar items (left-to-right) | Secondary icon, Page title, Back/Close icon | Back/Close icon, Page title, Secondary icon | Follow LTR |
| Ordering of buttons/actions in dialogs and sheets (left-to-right) | Left-aligned: Primary, Secondary | Right-aligned: Secondary, Primary | Follow RTL |

Am I missing any other layout use case?

**Comment by LittleAngelwings** [comment]
I think I miss a simply handle for left handed, like me, with LTR language. For me it will be better to have the ordering of toolbar items (left-to-right as follow): main menu, tab view and then the address bar. All other buttons with relevance wich were on right position, move them to the left side. Buttons on home screen, on new tab tray. The sliders and so on on settings to the left side, were a nice to have, but not a must be.

**Comment by saschanaz** [comment]
Is there a way to force RTL layout with LTR language?

**Comment by yoasif** [comment]
https://github.com/mozilla-mobile/fenix/issues/14750 seems relevant to me.

**Comment by regs01** [comment]
Is there any ETA for UI usability? I can't open tabs and I can't create a new tab, so I have to use Chrome.

**Comment by csadilek** [comment]
Moved to bugzilla: https://bugzilla.mozilla.org/show_bug.cgi?id=1812823

Change performed by the [Move to Bugzilla add-on](https://addons.mozilla.org/en-US/firefox/addon/move-to-bugzilla/).
